rpcs/wsFetcherService.go

```diff
@@ -104,7 +105,48 @@ func (fs *WsFetcherService) RequestBlock(ctx context.Context, target network.Uni
 		delete(fs.pendingRequests, waitKey)
 		fs.mu.Unlock()
 	}()
+	if target.Version() == "2.1" {
+		roundBin := make([]byte, binary.MaxVarintLen64)
+		binary.PutUvarint(roundBin, uint64(round))
+		topics := network.Topics{
+			network.MakeTopic(network.RequestDataTypeKey,
+				[]byte(network.BlockAndCertValue)),
+			network.MakeTopic(
+				network.RoundKey,
+				roundBin),
+		}
+		resp, err := target.Request(ctx, tag, topics)
+		if err != nil {
+			return WsGetBlockOut{}, fmt.Errorf("WsFetcherService.RequestBlock(%d): Request failed, %v", round, err)
+		}
+
+		if errMsg, found := resp.Topics.GetValue(network.ErrorKey); found {
+			return WsGetBlockOut{}, fmt.Errorf(string(errMsg))
```

Comment:
Avoid using `fmt.Errorf` if you're not going to add any information to the error. `errors.New` would do that for you.
In this case, you want to argument the error string with some context-related data, since the error is literally just the string that came from the other end.